Consul/Configuration.cs

```diff
@@ -1002,38 +1002,82 @@ public class ServiceResolverEntry : IConfigurationEntry
-        public string RequestTimeout { get; set; }
+        public Dictionary<string, ServiceResolverSubset> Subsets { get; set; }
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        public Dictionary<string, string> Subsets { get; set; }
+        public ServiceResolverRedirect Redirect { get; set; }
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        public string Filter { get; set; }
+        public Dictionary<string, ServiceResolverFailover> Failover { get; set; }
 
+        [JsonConverter(typeof(DurationTimespanConverter))]
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        public bool OnlyPassing { get; set; }
+        public TimeSpan? ConnectTimeout { get; set; }
 
+        [JsonConverter(typeof(DurationTimespanConverter))]
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        public string DefaultSubset { get; set; }
+        public TimeSpan? RequestTimeout { get; set; }
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        public Dictionary<string, string> Redirect { get; set; }
+        public LoadBalancerConfig LoadBalancer { get; set; }
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        public LoadBalancerConfig LoadBalancer { get; set; }
+        public Dictionary<string, string> Meta { get; set; }
+    }
+
+    public class ServiceResolverSubset
+    {
+        public string Filter { get; set; }
+        public bool OnlyPassing { get; set; }
+    }
+
+    public class ServiceResolverRedirect
+    {
+        public string Service { get; set; }
+        public string ServiceSubset { get; set; }
+        public string Namespace { get; set; }
+        public string Partition { get; set; }
+        public string Datacenter { get; set; }
+        public string Peer { get; set; }
+        public string SamenessGroup { get; set; }
+    }
+
+    public class ServiceResolverFailover
+    {
+        public string Service { get; set; }
+        public string ServiceSubset { get; set; }
+        // Referencing other partitions is not supported.
+        public string Namespace { get; set; }
+        public List<string> Datacenters { get; set; }
+        public List<ServiceResolverFailoverTarget> Targets { get; set; }
+        public ServiceResolverFailoverPolicy Policy { get; set; }
+        public string SamenessGroup { get; set; }
+    }
+
+    public class ServiceResolverFailoverTarget
+    {
+        public string Service { get; set; }
+        public string ServiceSubset { get; set; }
+        public string Partition { get; set; }
+        public string Namespace { get; set; }
+        public string Datacenter { get; set; }
+        public string Peer { get; set; }
+    }
+
+    public class ServiceResolverFailoverPolicy
+    {
+        public string Mode { get; set; }
+        public List<string> Regions { get; set; }
```

Comment:
Would it be possible to test these structs as well?
In the past we added some new structs without proper tests and they turned out to be incorrectly defined.

Reply:
@marcin-krystianc  So, I am working on the tests right now, and I also realized, that I have implimented ServiceResolverEntry according with the latests version of Consul API. That shouldn't harm backwards compitability, but that should provide some compatibility with future versions and less work in the future. Should I just leave it at that, or cut out the new fields to avoid confusion? Also, the Golang tests for the ServiceResolver class include some other test - for ServiceSplitter and ServiceRouter, should I write tests for them too, or just leave it at ServiceResolver? https://github.com/hashicorp/consul/blob/v1.9.17/api/config_entry_discoverychain_test.go . I will push the tests I currently have for it in a minute

> Would it be possible to test these structs as well? In the past we added some new structs without proper tests and they turned out to be incorrectly defined.

Reply:
> Should I just leave it at that, or cut out the new fields to avoid confusion?
Please keep the new fields in.